lib/collections.go

```diff
@@ -201,6 +201,25 @@ import (
 //	[1,2,3,4,5,6,7].min()  // return 1
 //	min([1,2,3,4,5,6,7])   // return 1
 //
+// # Tail
+//
+// Returns the elements of a list after the first element:
+//
+//	tail(<list<dyn>>) -> <optional<dyn>>
+//
+// Examples:
+//
+//	tail([1, 2, 3, 4, 5, 6])  // return [2, 3, 4, 5, 6]
+//	tail([6])                 // return []
+//	tail([])                  // return []
+//
+// The conjugate of tail, getting the first element, can be achieved
+// directly using list indexing, for example if a is [1, 2, 3, 4, 5, 6]
+// and b is []:
+//
+//	a[?0]  // return 1
+//	b[?0]  // return optional.none
```

Comment:
Initially I had it implemented as `head`, but later realised with optional types I can get exactly the same behaviour with this, so the doc seemed like it would be helpful.